### ISSUE TITLE
Toggle for showing comparisons by default

### ISSUE THREAD

**Issue by ericpgreen2** [rilldata/rill#3181]

**Comment by begelundmuller** [comment]
What do you think the YAML key(s) should be? Also, is it just a simple enabled/disabled boolean, or should it encompass the full configurability of a) comparing by time or by a dimension, b) comparing by last period vs. previous day vs. custom?

**Comment by cohenscottr** [comment]
i think having a simple enabled/disabled flag to start is appropriate and solves the primary issue. 

maybe `default_comparison_on` true/false? + @medriscoll for visibility

**Comment by ericpgreen2** [comment]
And it'd be nice to have symmetry with `default_time_range`, so maybe `default_comparison`

**Comment by begelundmuller** [comment]
I would suggest `enabled` instead of `on` since `on` is often used in other contexts for triggers. I like the symmetry of `default_comparison`.

It's similar effort on the backend to do it now or later. @ericpgreen2, what is your preference? And if doing more, what fields do we need to support?

**Comment by ericpgreen2** [comment]
Yeah, let's just get all the `default_comparison` YAML in place now (since we don't need to document it all until we account for it in the frontend). We should support selecting which dimension is to be compared, and, if that dimension is the time dimension, which time range.

Thoughts on:
```yaml
default_comparison:
    enabled: true
    dimension: time # or dim_1_id, dim_2_id
    time_range: P4W # any ISO8601 duration
```

In the `dimension` key, "time" could be an alias for the timestamp dimension ID? This would mirror the dashboard's comparison selection menu.

**Comment by AdityaHegde** [comment]
I feel like it would make sense to use time comparison when `dimension` is not set.

Also how do we show `last period`? Perhaps not providing `time_range` will default to `last period`?

**Comment by ericpgreen2** [comment]
On defaults:
- agree, the default `dimension` should be `time`
- agree, the default `time_range` should be `last period`

On `time_range`, good question:
- maybe we should match the labels that we have in the application: `last period`, `previous day/week/month/quarter/year`
- instead of `custom`, the user can provide a ISO8601 duration 

@AdityaHegde, @begelundmuller, what do you guys think of this?

**Comment by AdityaHegde** [comment]
Let's not add any labels identified in UI here. P1D, P1W, P1M etc can map to the existing labels. Other ones will be handled similar to default time range. Also `last period` is set by not providing `time_range` perhaps?

**Comment by ericpgreen2** [comment]
As discussed in Slack, noting that it should be possible to explicitly (not just implicitly) set `last period`

**Comment by AdityaHegde** [comment]
The setting for `default_comparison` has been merged. Adding additional support to configure exactly which time ranges and comparison settings for those will be done as a part of #3404

**Comment by chuck-alt-delete** [comment]
Hi! New user playing around and I have an issue related to this. I like that I can set `default_time_range`, but it seems that the comparison defaults to "Previous Period" which may not be working as intended. Here is a screenshot: 

<img width="1095" alt="Screenshot 2024-03-11 at 11 37 17" src="https://github.com/rilldata/rill/assets/87353672/9d6b295b-1257-419f-b49e-0c6749e19159">

The time period is Year to Date (`rill-YTD`), and I want the default comparison to be the Previous Year (`rill-PY`). It defaults to Previous Period. Notice the previous period starts in October 24 compared to January 1. I don't know if the previous period is actually supposed to be previous year in this case? I'd like to be able to customize the default comparison to use `rill-PY`.

Here is my config:
```yml
timeseries: ts
default_time_range: rill-YTD
default_comparison:
  mode: time
```
From what I can tell from the docs, there is no way to set the default comparison time range, only the initial `default_time_range`.

I'd appreciate any help on this. Thanks!

**Comment by cohenscottr** [comment]
hey Chuck  - did you also try the `comparison_offsets` setting? i _think_ that's what you're looking for? those are predefined ranges - but prior year would work.

<img width="871" alt="image" src="https://github.com/rilldata/rill/assets/82397988/de868f1f-a104-4e69-aec8-2f96b6cdeb6b">

**Comment by chuck-alt-delete** [comment]
Thanks, I’ll take a look!

**Comment by chuck-alt-delete** [comment]
Setting `comparison_offsets` didn't help. I think that overrides the allowed values for the comparison time range, but not the default selection. "Previous Period" was still the default after setting this:

```yml
available_time_ranges:
- P12M
- range: rill-YTD
  comparison_offsets:
    - rill-PY
```
